src/main/scala/me/rexim/morganey/interpreter/TermOutputHelper.scala

```diff
@@ -1,24 +1,41 @@
 package me.rexim.morganey.interpreter
 
 import me.rexim.morganey.ast.LambdaTerm
-import me.rexim.morganey.church.ChurchNumberConverter._
-import me.rexim.morganey.church.ChurchPairConverter._
+import me.rexim.morganey.meta._
+
+import scala.annotation.tailrec
 
 object TermOutputHelper {
 
-  def isPrintableChar(code: Int): Boolean =
-    32 <= code && code <= 176
-
-  def smartShowTerm(term: LambdaTerm): String = {
-    decodeNumber(term) match {
-      case Some(number) if isPrintableChar(number) => s"char: '${number.toChar}'"
-      case Some(number) => s"number: $number"
-      case None => decodeListOfNumbers(term) match {
-        case Some(numbers) if numbers.forall(isPrintableChar) =>
-          s"""string: \"${numbers.map(_.toChar).mkString}\""""
-        case Some(numbers) => s"numbers: [${numbers.mkString(",")}]"
-        case None => s"term: $term"
+  def smartShowTerm(term: LambdaTerm, noPrefix: Boolean = false): String = {
```

Comment:
I disagree. Consider `setVisible(visible: Boolean)`. It'd be a pain sometimes to use `show` + `hide` instead of `setVisible`.